[BUG] FreeBSD 14 - use of undeclared identifier 'get_current_dir_name' in i_system.cpp

### GZDoom version

Git

### Which game are you running with GZDoom?

None

### What Operating System are you using?

Other

### Please describe your specific OS version

FreeBSD 14.0-CURRENT amd64

### Relevant hardware info

_No response_

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

Trying to build gzdoom from source end up on an error

### Steps to reproduce the behaviour.

Simply build gzdoom on FreeBSD 14.0-CURRENT amd64

### Your configuration

```ini
cmake . -DCMAKE_BUILD_TYPE=Release -DZMUSIC_LIBRARIES=~/Downloads/ZMusic/source/libzmusic.so -DZMUSIC_INCLUDE_DIR=~/Downloads/ZMusic/include -DHAVE_VULKAN=OFF ; make
```


### Provide a Log

```
[ 18%] Linking C executable lemon
/home/Adrien2002/Downloads/gzdoom/build/tools/lemon
[ 18%] Built target lemon
[ 18%] Building C object tools/zipdir/CMakeFiles/zipdir.dir/zipdir.c.o
[ 19%] Linking C executable zipdir
[ 19%] Built target zipdir
[ 19%] Building C object libraries/gdtoa/CMakeFiles/arithchk.dir/arithchk.c.o
[ 19%] Linking C executable arithchk
[ 19%] Built target arithchk
[ 20%] Generating arith.h
[ 20%] Building C object libraries/gdtoa/CMakeFiles/qnan.dir/qnan.c.o
[ 20%] Linking C executable qnan
[ 20%] Built target qnan
[ 21%] Generating gd_qnan.h
[ 21%] Building C object libraries/gdtoa/CMakeFiles/gdtoa.dir/dmisc.c.o
[ 21%] Building C object libraries/gdtoa/CMakeFiles/gdtoa.dir/dtoa.c.o
[ 21%] Building C object libraries/gdtoa/CMakeFiles/gdtoa.dir/misc.c.o
/home/Adrien2002/Downloads/gzdoom/libraries/gdtoa/misc.c:70:50: warning: comparison of integers of different signs: 'long' and 'unsigned long' [-Wsign-compare]
                if (k <= Kmax && pmem_next - private_mem + len <= PRIVATE_mem) {
                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~
1 warning generated.
[ 21%] Linking C static library libgdtoa.a
[ 21%] Built target gdtoa
[ 21%] Building CXX object libraries/discordrpc/src/CMakeFiles/discord-rpc.dir/discord_rpc.cpp.o
[ 21%] Building CXX object libraries/discordrpc/src/CMakeFiles/discord-rpc.dir/rpc_connection.cpp.o
[ 21%] Building CXX object libraries/discordrpc/src/CMakeFiles/discord-rpc.dir/serialization.cpp.o
[ 22%] Building CXX object libraries/discordrpc/src/CMakeFiles/discord-rpc.dir/connection_unix.cpp.o
[ 22%] Building CXX object libraries/discordrpc/src/CMakeFiles/discord-rpc.dir/discord_register_linux.cpp.o
[ 22%] Linking CXX static library libdiscord-rpc.a
[ 22%] Built target discord-rpc
src/gitinfo.h updated to commit g4.11pre-178-g39cea95db.
[ 22%] Built target revision_check
[ 22%] Generating sc_man_scanner.h
[ 22%] Generating xlat_parser.c, xlat_parser.h
[ 22%] Generating zcc-parse.c, zcc-parse.h
[ 23%] Building CXX object src/CMakeFiles/zdoom.dir/posix/i_steam.cpp.o
[ 23%] Building CXX object src/CMakeFiles/zdoom.dir/common/platform/posix/i_system_posix.cpp.o
[ 23%] Building C object src/CMakeFiles/zdoom.dir/common/platform/posix/sdl/crashcatcher.c.o
[ 23%] Building CXX object src/CMakeFiles/zdoom.dir/common/platform/posix/sdl/hardware.cpp.o
[ 23%] Building CXX object src/CMakeFiles/zdoom.dir/common/platform/posix/sdl/i_gui.cpp.o
[ 23%] Building CXX object src/CMakeFiles/zdoom.dir/common/platform/posix/sdl/i_input.cpp.o
[ 23%] Building CXX object src/CMakeFiles/zdoom.dir/common/platform/posix/sdl/i_joystick.cpp.o
[ 24%] Building CXX object src/CMakeFiles/zdoom.dir/common/platform/posix/sdl/i_main.cpp.o
[ 24%] Building CXX object src/CMakeFiles/zdoom.dir/common/platform/posix/sdl/i_system.cpp.o
/home/Adrien2002/Downloads/gzdoom/src/common/platform/posix/sdl/i_system.cpp:411:17: error: use of undeclared identifier 'get_current_dir_name'
        char* curdir = get_current_dir_name();
                       ^
/home/Adrien2002/Downloads/gzdoom/src/common/platform/posix/sdl/i_system.cpp:450:17: error: use of undeclared identifier 'get_current_dir_name'
        char* curdir = get_current_dir_name();
                       ^
/home/Adrien2002/Downloads/gzdoom/src/common/platform/posix/sdl/i_system.cpp:454:3: error: use of undeclared identifier 'Printf'
                Printf("Opening folder: %s\n", infolder);
                ^
/home/Adrien2002/Downloads/gzdoom/src/common/platform/posix/sdl/i_system.cpp:460:3: error: use of undeclared identifier 'Printf'
                Printf("Unable to open directory '%s\n", infolder);
                ^
4 errors generated.
*** Error code 1

Stop.
make[2]: stopped in /usr/home/Adrien2002/Downloads/gzdoom/build
*** Error code 1

Stop.
make[1]: stopped in /usr/home/Adrien2002/Downloads/gzdoom/build
```

--- a/src/common/platform/posix/sdl/i_system.cpp
+++ b/src/common/platform/posix/sdl/i_system.cpp
@@ -51,7 +51,6 @@
 #include <asm/unistd.h>
 #include <linux/perf_event.h>
 #include <sys/mman.h>
-#include "printf.h"
 #endif
 
 #include <SDL.h>
@@ -65,6 +64,7 @@
 #include "c_cvars.h"
 #include "palutil.h"
 #include "st_start.h"
+#include "printf.h"
 
 
 #ifndef NO_GTK
@@ -408,7 +408,7 @@ FString I_GetFromClipboard (bool use_primary_selection)
 
 FString I_GetCWD()
 {
-	char* curdir = get_current_dir_name();
+	char* curdir = getcwd(NULL,0);
 	if (!curdir) 
 	{
 		return "";
@@ -447,7 +447,8 @@ unsigned int I_MakeRNGSeed()
 
 void I_OpenShellFolder(const char* infolder)
 {
-	char* curdir = get_current_dir_name();
+
+	char* curdir = getcwd(NULL,0);
 
 	if (!chdir(infolder))
 	{

